packages/@aws-cdk/pipelines/lib/main/pipeline.ts

```diff
@@ -21,49 +21,44 @@ export interface PipelineProps {
    * may need to configure it explicitly.
    */
   readonly synth: Step;
-
-  /**
-   * The deployment engine
-   *
-   * This controls what deployment system will be used to deploy this
-   * pipeline.
-   *
-   * Right now, only `new CodePipelineEngine()` is supported in this library,
-   * though 3rd implementations may be available.
-   *
-   * The engine is invoked when `buildPipeline()` is called, or when
-   * `app.synth()` is called (whichever happens first).
-   */
-  readonly engine: IDeploymentEngine;
 }
 
 /**
  * A generic CDK Pipelines pipeline
  *
- * The Pipeline should receive an Engine which builds actual infrastructure from
- * the pipeline definition.
+ * Different deployment systems will provide subclasses of `Pipeline` that generate
+ * the deployment infrastructure necessary to deploy CDK apps, specific to that system.
+ *
+ * This library comes with the `CodePipeline` class, which uses AWS CodePipeline
+ * to deploy CDK apps.
  *
  * The actual pipeline infrastructure is constructed (by invoking the engine)
  * when `buildPipeline()` is called, or when `app.synth()` is called (whichever
  * happens first).
  */
-export class Pipeline extends CoreConstruct {
-  private readonly blueprint: Blueprint;
-  private readonly engine: IDeploymentEngine;
+export abstract class Pipeline extends CoreConstruct {
```

Comment:
Per convention we usually add `Base` postfix to abstract base classes: `PipelineBase`